chia/cmds/db_upgrade_func.py

```diff
@@ -237,22 +237,22 @@ async def convert_v1_to_v2(in_path: Path, out_path: Path) -> None:
             commit_in = HINT_COMMIT_RATE
             hint_start_time = time()
             hint_values = []
-            await out_db.execute("CREATE TABLE hints(id INTEGER PRIMARY KEY AUTOINCREMENT, coin_id blob,  hint blob)")
+            await out_db.execute("CREATE TABLE IF NOT EXISTS hints(coin_id blob, hint blob, UNIQUE (coin_id, hint))")
             await out_db.commit()
             async with in_db.execute("SELECT coin_id, hint FROM hints") as cursor:
```

Comment:
is there a way to check that upgrade has been finished successfully? (worry being if you start and stop full node before upgrade has been performed table will exist but it will be incomplete)

Reply:
the upgrade doesn't happen in-place, so to a new DB file. And if you "restart" if will fail because the target file already exists. There's no explicit record of the conversion completing successfully, maybe there should be. The problem you describe is not limited to the hints table though

Reply:
you say "start and stop full node", the conversion doesn't happen inside the full node, it's done as a manual command:
```
chia db upgrade
```

Reply:
I think I see what you mean. This should not say "IF NOT EXISTS". That table should never exist here, if it does, something's wrong and we need to fail. Fixed.